modules/upgrade/pages/migrate-3x-4x.adoc

```diff
@@ -1,3 +1,445 @@
+14:58:29   Importing database dump. Please wait...
+15:05:11   Database dump successfully imported.
+----
+
+
+For this example dumping the database takes around five minutes to complete.
+Importing the dump onto the target system will take an additional seven minutes.
+For big installations this can take up to several hours.
+You should also account for the time it takes to copy all the package data to the new machine.
+Depending on your network infrastructure and hardware, this can also take a significant amount of time.
+
+
+
+[[bp.sec.mgr.migration.speedup]]
+== Speeding up the Migration
+
+A complete migration can consume a lot of time.
+This is caused by the amount of data that must be copied.
+Total migration time can be greatly decreased by eliminating the need to copy data prior to performing the migration (for example, channels, packages, auto-install images, and any additional data).
+You can gather all data via {yast} by running:
+
+----
+mgr-setup -r
+----
+
+Executing [command]``mgr-setup -r`` will copy the data from the old server to the new one.
+This command may be run at any time and your current server will remain fully functional.
+When the migration has been initiated only data changed since running [command]``mgr-setup -r`` will need to be transferred.
+This significantly reduces downtime.
+
+On large installations transfering the database (which involves dumping the database onto the source machine and then importing the dump onto the target system) will still take some time.
+During the database transfer no write operations should occur therefore the migration script will shut down any {productname} database services running on the source machine.
+
+
+
+[[bp.sec.mgr.migration.pkg.extern]]
+== Packages on External Storage
+
+
+Some installations may store the package data on external storage (for example, NFS mount on [path]``/var/spacewalk/packages``).
+You do not need to copy this data to the new machine.
+Edit the script located in [path]``/usr/lib/susemanager/bin/mgr-setup`` and remove the respective [command]``rsync`` command (located around line 442).
+
+.Mounting External Storage
+[IMPORTANT]
+====
+Make sure your external storage is mounted on the new machine before starting the system for the first time.
+Analogue handling for [path]``/srv/www/htdocs/pub`` if appropriate.
+
+In general, all needed files and directories, not copied by the migration tool, should be copied to the new server manually.
+====
+
+
+
+[[bp.sec.mgr.migration.trouble]]
+== Troubleshooting
+
+=== Broken {webui} after Migration
+
+It is possible that the {webui} may break during migration.
+This behavior is not a bug, but a browser caching issue.
+The new machine has the same host name and IP address as the old machine.
+This duplication can confuse some Web browsers.
+If you experience this issue reload the page.
+For example, in Firefox pressing the key combination kbd:[Ctrl+F5] should resume normal functionality.
```

Comment:
These should probably be re-written in CCFR format, but I'll leave that up to you ;)

Reply:
@Loquacity "CCFR format"? I cannot solve this riddle ;-(

Reply:
Sorry!  Cause-Consequence-Fix-Result. That's the format I implemented in the Troubleshooting section of the Admin Guide:

> One sentence each:
> Cause: What created the problem?
> Consequence: What does the user see when this happens?
> Fix: What can the user do to fix this problem?
> Result: What happens after the user has completed the fix?
> 
> If more detailed instructions are required, put them in a "Resolving" procedure:
> .Procedure: Resolving Widget Wobbles
> . First step
> . Another step
> . Last step

Reply:
ok, fair request.  Nevertheless, I'll probably postpone it a little bit.

Reply:
Yeah, that's fair enough.